lua/toggleterm.lua

```diff
@@ -110,11 +110,22 @@ end
 function M.handle_term_enter()
   local _, term = terms.identify()
   if term then
-    term:resize()
+    local ui = require("toggleterm.ui")
+    local persist_size = require("toggleterm.config").get("persist_size")
+    if term:is_split() and not (persist_size and ui.persistent[term.direction]) then
```

Comment:
Here is another example where I wonder if the specifics of `persistent` can't be encapsulated inside of resize_split i.e. can that function do the checking if it should bother persisting sizes or not then this module doesn't have to deal with directly checking the persistence table